### PR TITLE
Theme upgrade modals: Change tier names

### PR DESCRIPTION
**Pull request by dsas**
Make some of the theme tier names align with the plans required for them and upgrade modals to match.

Related to #86482

## Proposed Changes

Change the tier names for the personal and premium plans in the modal. I've purposefully changed the heading to "Unlock this theme" so that it fits on one line which looks nicer and also to reduce the repetition of Starter/Explorer everywhere.

I also considered making the Explorer themes modal list "Starter themes" in the plan benefits, but thought maybe that's just too confusing and isn't exactly related to the aim of "changing tier names".

### Modals
1. Use the calypso live link
2. Go through to the design picker using a free plan
3. Reload the page after appending flags=themes/tiers to the url
4. Choose a Starter theme like Epi, press "Unlock theme"
5. Choose an Explorer theme like Annalee, press "Unlock theme"

Theme type | Before | After
----|---|------
Starter | Screenshot 2024-01-17 at 09 26 06  | Screenshot 2024-01-17 at 11 53 46
Explorer | Screenshot 2024-01-17 at 09 32 48  | Screenshot 2024-01-17 at 11 53 09

### Plans grid
1. Use the calypso live link
2. Go to /plans/:siteslug

Before | After
-------|------
Screenshot 2024-01-17 at 12 02 15 |Screenshot 2024-01-17 at 12 01 42

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/lib/plans/features-list.tsx
Comment (dsas) [inline client/lib/plans/features-list.tsx]: This change will affect the the plans grid. 

I haven't changed the title/description as it's using premium in a more descriptive sense and is used to supplement the "Explorer themes" label, e.g. in alt text.

File: client/lib/plans/features-list.tsx
Comment (dsas) [inline client/lib/plans/features-list.tsx:547]: I don't want to block this PR on it, but the business mention surprised me. I don't think the Starter plan is generally the best fit there. Something for another day.

File: client/components/theme-upgrade-modal/index.tsx
Comment (rcrdortiz) [inline client/components/theme-upgrade-modal/index.tsx:128]: Nice that we're going with generic reference instead of explicit, will save us time if we ever change plan names/tiers again.

File: client/components/theme-upgrade-modal/index.tsx
Comment (rcrdortiz) [inline client/components/theme-upgrade-modal/index.tsx]: I'm seeing that this relies on `plan: plans.data?.[ PLAN_PERSONAL ]?.productNameShort || ''` at line 135.
Do you know the difference between that and `getPlan( PLAN_PERSONAL )?.getTitle()`.

Should we try to always use the same approach?

File: client/components/theme-upgrade-modal/index.tsx
Comment (rcrdortiz) [inline client/components/theme-upgrade-modal/index.tsx]: Same as my other comment about how we fetch the plan name.

File: client/lib/plans/features-list.tsx
Comment (rcrdortiz) [reply client/lib/plans/features-list.tsx:547]: Yup, the business reference is strange. 
Should we implement something so that we can retrieve the `{you name it} themes` as we do with the plan names? That way we would only need to change the `getTitle` implementation instead of searching and replacing all occurences. We don't need to do this in this PR, we could do it as boyscout work. 
WDYT?

Or couldn't we just replace this with a similar approach to the one we use on line 172?
```
'Get access to our %(premiumPlanName)s themes, and a...
```

File: client/lib/plans/features-list.tsx
Comment (rcrdortiz) [reply client/lib/plans/features-list.tsx]: If this affects the Plans Grid, I think that we should add the `String Freeze` label to this PR. PaYJgx-3sX-p2

File: client/lib/plans/features-list.tsx
Comment (dsas) [reply client/lib/plans/features-list.tsx:547]: The difference is case, it would say "Get access to our Starter themes", which might be the right thing to do, but isn't what we were doing before.

I think we should probably put this PR on hold until paYJgx-4vW-p2 is resolved.

File: client/components/theme-upgrade-modal/index.tsx
Comment (dsas) [reply client/components/theme-upgrade-modal/index.tsx]: `usePlans` (where `plans` here comes from) uses a rest endpoint. `getPlan` is backed by the `PLANS_LIST` constant.

I've just tested this modal on live preview and it appears to be broken in a similar way to #86411 though, so I guess it needs to use getPlan.

File: client/lib/plans/features-list.tsx
Comment (dsas) [reply client/lib/plans/features-list.tsx]: yeah I was anticipating some feedback on the strings so I was holding off, and lo and behold!

Comment (rcrdortiz) [reply client/lib/plans/features-list.tsx]: When we merge these changes the Plans Grid will show the `Starter Themes` feature regardless of if we shipped the new tier or not. I think it's already happening in production though.

File: client/components/theme-upgrade-modal/index.tsx
Comment (dsas) [reply client/components/theme-upgrade-modal/index.tsx]: Even though with the current string nothing has changed, it's still broken on my sandbox and working on prod.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10826520

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @dsas for including a screenshot in the description! This is really helpful for our translators.

**Review by rcrdortiz — APPROVED** [review]
Besides that this will be available to everyone once merged, everything is working as expected.

File: client/lib/plans/features-list.tsx
Comment (dsas) [reply client/lib/plans/features-list.tsx]: We don't include Starter themes in the plans grid with or without this change. We do include "Premium themes" and that will change to "Explorer themes" with this change.

We can revaluate when translations are ready whether we need to adjust before deploying.

File: client/lib/plans/features-list.tsx
Comment (dsas) [inline client/lib/plans/features-list.tsx]: this breaks in live previews (wpcom-block-editor) with "Could not find config value for key 'english_locales'"

I'll just remove it and rely on the feature flag.

File: client/components/theme-upgrade-modal/index.tsx
Comment (dsas) [reply client/components/theme-upgrade-modal/index.tsx]: Opened https://github.com/Automattic/wp-calypso/pull/86579 to deal with this - it's broken on trunk too.

**Comment by dsas** [comment]
Dropped the string freeze tag, since the plans grid change is now behind the hasTranslation check and many of the other changes are behind feature flags.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10826520).

**Pull request merged**
